src/envoy/mixer/quota_config.h

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "include/attribute.h"
+#include "mixer/v1/config/client/quota.pb.h"
+
+namespace Envoy {
+namespace Http {
+namespace Mixer {
+
+// An object uses quota config to check a request
+// attributes to generate required quotas.
+class QuotaConfig {
+ public:
+  QuotaConfig(const ::istio::mixer::v1::config::client::QuotaSpec& spec_pb);
+
+  struct Quota {
+    std::string quota;
+    int64_t charge;
+
+    // For unit-test.
```

Comment:
If this is for unit test only, just add a matcher for gtest.

Reply:
It is used in 
ASSERT_EQ for std::vector<Quota>
gTest matcher will not help.

Reply:
std::vector of Quota